src/main/java/consumers/Consumer.java

```diff
@@ -0,0 +1,36 @@
+package consumers;
+
+import brokers.Broker;
+import messages.Message;
+import topics.Topic;
+
+public interface Consumer {
+    /**
+     * Initializes the consumer by communicating with a broker in the cluster.
+     * The broker assignment for the specified topic and id is obtained during this process.
+     *
+     * @param broker The initial broker to communicate with.
+     * @param topic The topic from which messages will be consumed.
+     * @param id The driver/rider id we are interested in OR -1 if not needed
+     */
+    void initialize(Broker broker, Topic topic, int id);
```

Comment:
it's the produceRequest from fig. 3 in our design doc. So we initialize our consumer and it gets assigned a broker. I'm not sure if this means we will create a new consumer for every topic or not

Reply:
> Oh wait are these in case we communicate with the broker through java?

Ohhhh I didn't realize what you meant. I guess we don't need consumer and producer classes then.